src/main/java/util/RefreshTimer.java

```diff
@@ -0,0 +1,101 @@
+package util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * Facilitates the countdown for periodic refreshes.
+ */
+public class RefreshTimer extends TickingTimer{
```

Comment:
@damithc I'm thinking whether we should put the entire codes related to refreshing to the ApiQuotaManager class. For example, the updateSyncRefreshRate method in GUIController. We will register the RefreshRateTriggeredEvent in this class. However it will implicitly coupled to the Logic class, as it will need to call the refresh() in the Logic class.